src/core/qgsvectorlayer.cpp

```diff
@@ -4445,3 +4452,27 @@ bool QgsVectorLayer::readExtentFromXml() const
 {
   return mReadExtentFromXml;
 }
+
+void QgsVectorLayer::setRefreshOnNotifyEnabled( bool enabled )
+{
+  if ( !dataProvider() )
+    return;
+
+  if ( enabled and not isRefreshOnNotifyEnabled() )
+  {
+    connect( dataProvider(), &QgsVectorDataProvider::notify, this, &QgsVectorLayer::onNotified );
+  }
+  else if ( !enabled and isRefreshOnNotifyEnabled() )
+  {
+    disconnect( dataProvider(), &QgsVectorDataProvider::notify, this, &QgsVectorLayer::onNotified );
+  }
+  dataProvider()->setListening( enabled );
+  mIsRefreshOnNofifyEnabled = enabled;
+}
+
+void QgsVectorLayer::onNotified( const QString &message )
+{
+  if ( refreshOnNotifyMessage().isEmpty() or refreshOnNotifyMessage() == message )
+    triggerRepaint();
```

Comment:
Should that emit a `dataChanged()` signal so other parts than only the canvas can also benefit from the signal?

Reply:
In this case, the same should be done with the mRefreshTimer. What do you think ?

Reply:
Possibly... Although not sure since the refresh timer doesn't actually check for any changes, I'm not sure I want to attribute table reloaded every 5 seconds "just because".